core/core-impl/src/main/java/ai/timefold/solver/core/impl/solver/DefaultSolverJobSession.java

```diff
@@ -0,0 +1,82 @@
+package ai.timefold.solver.core.impl.solver;
+
+import java.util.Objects;
+import java.util.UUID;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import ai.timefold.solver.core.api.domain.solution.PlanningSolution;
+import ai.timefold.solver.core.api.solver.SolverJob;
+import ai.timefold.solver.core.api.solver.SolverJobBuilder;
+import ai.timefold.solver.core.api.solver.SolverManager;
+import ai.timefold.solver.core.config.solver.SolverJobConfig;
+import ai.timefold.solver.core.config.solver.termination.TerminationConfig;
+
+/**
+ * A {@link SolverManager} can solve multiple planning problems and can be used across different threads. The session of
+ * solver manager is responsible for constructing and solving a specific configuration problem. Hence, it is possible to
+ * have multiple distinct sessions that are scheduled to run by the SolverManager implementation.
+ * <p>
+ * To solve a planning problem, set the problem configuration: {@link #withProblemId(Object)},
+ * {@link #withProblemFinder(Function)} and {@link #withProblem(Object)}. Then solve it by calling {@link #run()}.
+ *
+ * @param <Solution_> the solution type, the class with the {@link PlanningSolution} annotation
+ * @param <ProblemId_> the ID type of submitted problem, such as {@link Long} or {@link UUID}.
+ *
+ * @see DefaultSolverManager
+ */
+public class DefaultSolverJobSession<Solution_, ProblemId_> implements SolverJobBuilder<Solution_, ProblemId_> {
```

Comment:
If you're implementing `SolverJobBuilder`, typically we call the implementation `DefaultSolverJobBuilder`.

In this case, it is questionable whether we need the interface at all - why not put the make this the `SolverJobBuilder` directly? Make the constructors package-private and there are no issues.

Reply:
> If you're implementing `SolverJobBuilder`, typically we call the implementation `DefaultSolverJobBuilder`.

Done!

> In this case, it is questionable whether we need the interface at all - why not make this the `SolverJobBuilder` directly? Make the constructors package-private and there are no issues.

The interface belongs to the public API, and `DefaultSolverJobBuilder` needs to access `DefaultSolverManager` in order to submit the problem. So, it seems not correct to access `DefaultSolverManager` from the public API. Does that make sense?

Reply:
I see your point. Technically, you're correct - it creates dependency from public API to private code, and it would be cleaner to separate the API from the impl as you've done.

That said, this (anti)pattern is already *everywhere* in the Solver, and often it can not even be avoided. Therefore I don't mind if we do it here as well. I don't mind if you want to keep it like that, but in that case I see there is a very long Javadoc on the class which should only be on the interface, as the user-facing description. (Javadocs on the implementation should only deal with the details of the implementation.)